src/components/views/rooms/UserIdentityWarning.tsx

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
+Please see LICENSE files in the repository root for full details.
+*/
+
+import React from "react";
+import {
+    CryptoEvent,
+    EventType,
+    KnownMembership,
+    MatrixEvent,
+    Room,
+    RoomStateEvent,
+    RoomMember,
+} from "matrix-js-sdk/src/matrix";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import type { CryptoApi, UserVerificationStatus } from "matrix-js-sdk/src/crypto-api";
+import { _t } from "../../../languageHandler";
+import AccessibleButton, { ButtonEvent } from "../elements/AccessibleButton";
+import MemberAvatar from "../avatars/MemberAvatar";
+import { MatrixClientPeg } from "../../../MatrixClientPeg";
+import { SDKContext } from "../../../contexts/SDKContext";
+
+interface IProps {
```

Comment:
The typescript interface should not be marked with an `uppercase` I.

```suggestion
interface UserIdentityWarningProps {
```